osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/security/SetEncryptionKeyExchangeOnGMeterCommandExecutor.java

```diff
@@ -123,10 +125,8 @@ public MethodResultCode execute(final DlmsConnectionManager conn, final DlmsDevi
 
             return MethodResultCode.SUCCESS;
         } catch (final IOException e) {
-            LOGGER.error("Unexpected exception while connecting with device", e);
             throw new ConnectionException(e);
         } catch (final EncrypterException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)